src/Common/src/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -243,13 +375,8 @@ public bool IsMarshallingRequired()
         private TypeDesc GetNativeTypeFromMarshallerKind(TypeDesc type, MarshallerKind kind, MarshallerKind elementMarshallerKind,
                 MarshalAsDescriptor marshalAs)
         {
-            if (type.IsByRef)
-            {
-                return GetNativeTypeFromMarshallerKind(((ByRefType)type).ParameterType, kind, elementMarshallerKind, marshalAs).MakePointerType();
-            }
-
             TypeSystemContext context = PInvokeMethodData.Context;
-            NativeType nativeType = NativeType.Invalid;
+            NativeType nativeType = TypeSystem.NativeType.Invalid;
```

Comment:
@shrah hmm... most likely VS did this for me so I didn't realize introducing NativeType property in Marshaller class actually lead to those conflicts... :) We probably should just rename TypeSystem.NativeType to NativeTypeKind instead. 
@MichalStrehovsky any objections?